src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -625,6 +625,69 @@ private static string UnescapeDriveColon(string fileUri)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Converts a file system path into a DocumentUri required by Language Server Protocol.
+        /// </summary>
+        /// <remarks>
+        /// When sending a document path to a LSP client, the path must be provided as a
+        /// DocumentUri in order to features like the Problems window or peek definition
+        /// to be able to open the specified file.
+        /// </remarks>
+        /// <param name="path">
+        /// A file system path. Note: if the path is already a DocumentUri, it will be returned unmodified.
+        /// </param>
+        /// <returns>The file system path encoded as a DocumentUri.</returns>
+        internal static string ConvertPathToDocumentUri(string path)
+        {
+            const string fileUriPrefix = "file:///";
+
+            if (path.StartsWith("untitled:", StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (path.StartsWith(fileUriPrefix, StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // On a Linux filesystem, you can have multiple colons in a filename e.g. foo:bar:baz.txt
+                string absoluteUri = new Uri(path).AbsoluteUri;
+
+                // First colon is part of the protocol scheme, see if there are other colons in the path
+                int firstColonIndex = absoluteUri.IndexOf(':');
+                if (absoluteUri.IndexOf(':', firstColonIndex + 1) >= 0)
```

Comment:
nit: it's just slightly more correct because we expect it to be an index greater than firstColonIndex.
```suggestion
                if (absoluteUri.IndexOf(':', firstColonIndex + 1) > firstColonIndex)
```